File: src/main/java/com/example/kakebo/Repository/IngresoRepository.java
package com.example.kakebo.Repository;

import com.example.kakebo.entity.Ingreso;

import javax.enterprise.context.ApplicationScoped;
import javax.inject.Inject;
import javax.persistence.EntityManager;
import javax.persistence.EntityTransaction;
import javax.persistence.criteria.CriteriaQuery;
import javax.transaction.Transactional;
import javax.ws.rs.*;
import javax.ws.rs.core.Response;

@ApplicationScoped
@Path("/ingreso")
@Produces("application/json")
@Consumes("application/json")
public class IngresoRepository {

  @Inject
  private KakeboJPAFactory kjf;

  @GET
  public Response getAll() {
    EntityManager em = kjf.getEntityManager();
    CriteriaQuery<Ingreso> cq = em.getCriteriaBuilder().createQuery(Ingreso.class);
    return Response.ok(em.createQuery(cq.select(cq.from(Ingreso.class))).getResultList()).build();
  }
  @GET
  @Path("{id}")
  public Response getOne(@PathParam("id") long id){
    EntityManager em = kjf.getEntityManager();
    return Response.ok(em.find(Ingreso.class, id)).build();
  }

  @POST
  @Transactional
  public Response create(Ingreso ing) {
    EntityManager em = kjf.getEntityManager();
    EntityTransaction tx = em.getTransaction();
    tx.begin();
    em.persist(ing);
    tx.commit();
    return Response.ok(ing).build();
  }

  @DELETE
  @Transactional
  @Path("{id}")
  public Response delete(@PathParam("id") long id) {
    EntityManager em = kjf.getEntityManager();
    Ingreso ing = em.find(Ingreso.class, id);
    EntityTransaction tx = em.getTransaction();
    tx.begin();
    em.remove(ing);
    tx.commit();
    return Response.ok().build();
  }

  @PUT
  @Transactional
  @Path("{id}")
  public Response update(@PathParam("id") long id, Ingreso ing) {
    EntityManager em = kjf.getEntityManager();
    Ingreso in = em.find(Ingreso.class, id);
    EntityTransaction tx = em.getTransaction();
    tx.begin();
    in.setFin(ing.getFin());
    in.setInicio(ing.getInicio());
    in.setMonto(ing.getMonto());
    tx.commit();
    return Response.ok(in).build();
  }
}

File: src/main/kotlin/org/example/kakebo/repository/IngresoRepository.kt
package org.example.kakebo.repository

import io.quarkus.hibernate.orm.panache.PanacheRepository
import org.example.kakebo.entity.Ingreso
import javax.enterprise.context.ApplicationScoped
import javax.transaction.Transactional
import javax.ws.rs.*
import javax.ws.rs.core.MediaType

@ApplicationScoped
@Path("ingreso")
@Produces(MediaType.APPLICATION_JSON)
@Consumes(MediaType.APPLICATION_JSON)
class IngresoRepository: PanacheRepository<Ingreso> {

    @GET
    fun getIngreso(): List<Ingreso>? = listAll()

    @GET
    @Path("{id}")
    fun getIngresoById(@PathParam("id") id:Long): Ingreso? = findById(id)

    @POST
    @Transactional
    fun createIngreso(ingreso:Ingreso) {
        val ing = Ingreso()
        ing.fin = ingreso.fin
        ing.inicio = ingreso.inicio
        ing.monto = ingreso.monto
        persistAndFlush(ing)
    }

    @DELETE
    @Path("{id}")
    @Transactional
    fun deleteIngreso(@PathParam("id") id:Long) = deleteById(id)

    @PUT
    @Path("{id}")
    @Transactional
    fun updateIngreso(@PathParam("id") id:Long, ingreso:Ingreso) =
        update("fin=?1, inicio=?2, monto=?3 where id = ?4",ingreso.fin, ingreso.inicio, ingreso.monto, id )

}